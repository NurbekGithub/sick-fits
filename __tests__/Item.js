import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 5000,
  description: 'This item is soo cool',
  image: 'dog.png',
  largeImage: 'largeDog.png'
};

describe('<Item />', () => {
  const wrapper = shallow(<ItemComponent item={fakeItem} />);
  it('renders and matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  // it('renders and displays properly', () => {
  //   const PriceTag = wrapper.find('PriceTag');
  //   expect(PriceTag.children().text()).toBe('$50');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });
  // it('renders img properly', () => {
  //   expect(wrapper.find('img').props()).toEqual({src: fakeItem.image, alt: fakeItem.title})
  // });

  // it('renders out the buttons properly', () => {
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('AddToCart')).toHaveLength(1);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  //   console.log(buttonList.debug())
  // })
});