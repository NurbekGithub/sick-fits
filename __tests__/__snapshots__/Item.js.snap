// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.png"
  />
  <Title>
    <Link
      href="/item?id=ABC123"
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is soo cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href="/update?id=ABC123"
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    >
      Add To Cart ðŸ›’
    </AddToCart>
    <DeleteItem
      id="ABC123"
    >
      Delete item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
